src/procedures/procedure.c

```diff
@@ -22,18 +22,28 @@ static void _procRegister(const char *procedure, ProcGenerator gen) {
 // Register procedures.
 void Proc_Register() {
 	__procedures = raxNew();
-	_procRegister("db.labels", Proc_LabelsCtx);
-	_procRegister("db.propertykeys", Proc_PropKeysCtx);
-	_procRegister("db.relationshiptypes", Proc_RelationsCtx);
+	char proc_name_lower[128] = {0};
+	short proc_name_lower_len = 128;
+    str_tolower("db.labels", &proc_name_lower[0], &proc_name_lower_len);
+	_procRegister(proc_name_lower, Proc_LabelsCtx);
+    str_tolower("db.propertyKeys", &proc_name_lower[0], &proc_name_lower_len);
+    _procRegister(proc_name_lower, Proc_PropKeysCtx);
+    str_tolower("db.relationshipTypes", &proc_name_lower[0], &proc_name_lower_len);
+	_procRegister(proc_name_lower, Proc_RelationsCtx);
```

Comment:
the conversion to lower case should be done within `_procRegister`. much easier and cleaner.

Reply:
I agree, but in this case the function should return the new string instead of void right? so should I add another function or change the code in every place that uses str_tolower?

Reply:
Something along these lines:

```
_procRegister(proc_name, proc) {
char proc_name_lower[128];
uint proc_name_lower_len ;
str_tolower(proc_name, &proc_name_lower, &proc_name_lower_len);
...
}
```